### PR TITLE
UI: Change logic when edit button should be shown

### LINKED ISSUE #395: UI: Edit button is enabled although user hasn't permission to edit
### How to test

1. Grant a user only view permission.
2. Login with that user and view the resource.
3. It shouldn't be able to view nor click the edit button. Currently it is possible to view/click the edit button, but it's not possible to edit on submit.

### Solution

Permissions are only fetched on permissions dialog open. Maybe they should be fetched on overview/detailview screen.

### PR DESCRIPTION
**Pull request by Stezido**
## Description
It shouldn't be able to view nor click the edit button without update permissions. Currently it is possible to view/click the edit button, but it's not possible to edit on submit.

## Solution

- [x] Change view logic of project and subproject edit action button
- [x] Create an e2e-test

Closes #395

### CODE REVIEW

File: e2e-test/cypress/integration/project_edit_spec.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    cy.login();
+    cy.fetchProjects().then(p => (projects = p));
+  });
+
+  beforeEach(function() {
+    cy.login();
+    cy.visit(`/projects`);
+  });
+
+  it("Editing the title is possible", function() {
```
Comment (mathiashoeld) [inline e2e-test/cypress/integration/project_edit_spec.js]: add some comments to let the reader know what's happening. 
e.g. "getting value of text field" for line 23, "resetting the title" for line 25

File: e2e-test/cypress/integration/project_edit_spec.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      .invoke("val")
+      .then(title => {
+        cy.get("[data-test=nameinput] input").type("-");
+        cy.get("[data-test=submit]").should("be.enabled");
+        cy.get("[data-test=nameinput] input")
+          .clear()
+          .type(title);
+        cy.get("[data-test=submit]").should("be.disabled");
+      });
+  });
```
Comment (mathiashoeld) [inline e2e-test/cypress/integration/project_edit_spec.js]: newline

File: e2e-test/cypress/integration/project_edit_spec.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+describe("Project Edit", function() {
+  let projects = undefined;
+
+  before(() => {
+    cy.login();
+    cy.fetchProjects().then(p => (projects = p));
```
Comment (mathiashoeld) [inline e2e-test/cypress/integration/project_edit_spec.js]: this requires projects to be added. to make it independent of provisioning we should create the project for testing here

File: e2e-test/cypress/integration/subproject_edit_spec.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    cy.get("[data-test=nameinput] input")
+      .invoke("val")
+      .then(title => {
+        cy.get("[data-test=nameinput] input").type("-changed");
+        cy.get("[data-test=submit]").click();
+        cy.get("[data-test=subproject-title-0]")
+          .invoke("text")
+          .should("not.eq", title);
+      });
+  });
```
Comment (mathiashoeld) [inline e2e-test/cypress/integration/subproject_edit_spec.js]: newline

File: e2e-test/cypress/integration/subproject_edit_spec.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      .invoke("val")
+      .then(title => {
+        cy.get("[data-test=nameinput] input").type("-");
+        cy.get("[data-test=submit]").should("be.enabled");
+        cy.get("[data-test=nameinput] input")
+          .clear()
+          .type(title);
+        cy.get("[data-test=submit]").should("be.disabled");
+      });
+  });
```
Comment (mathiashoeld) [inline e2e-test/cypress/integration/subproject_edit_spec.js]: newline

File: e2e-test/cypress/integration/subproject_edit_spec.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  it("The edit button isn't visible without edit permissions", function() {
+    cy.server();
+    cy.get("[data-test=subproject-edit-button-0]").should("be.enabled");
+    cy.login("root", "root-secret");
+    cy.revokeSubprojectPermission(projectId, subprojectId, "subproject.update", "mstein");
+    cy.login();
+    cy.visit(`/projects/${projectId}`);
+    cy.get("[data-test=subproject-edit-button-0]").should("have.css", "opacity", "0");
```
Comment (mathiashoeld) [inline e2e-test/cypress/integration/subproject_edit_spec.js]: is the button also disabled? if not, this would be a serious bug

File: e2e-test/cypress/integration/workflowitem_edit_spec.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-describe("Workflowitem edit", function() {
+describe("Workflowitem edit", function () {
```
Comment (mathiashoeld) [inline e2e-test/cypress/integration/workflowitem_edit_spec.js]: formatting 
(issues on multiple lines, i'll just write the comment once)

File: e2e-test/cypress/integration/workflowitem_edit_spec.js [hunk 8]
```diff
@@ -113,33 +113,3 @@ describe("Workflowitem edit", function() {
     }
   );
 });
-
-describe("Workflowitem create", function() {
```
Comment (mathiashoeld) [inline e2e-test/cypress/integration/workflowitem_edit_spec.js]: why was this deleted?

Comment (Stezido) [reply e2e-test/cypress/integration/workflowitem_edit_spec.js]: We already tested it in workflowitem_create_spec. It's duplicate.

File: e2e-test/cypress/integration/subproject_edit_spec.js (hunk 6 above)
Comment (Stezido) [reply e2e-test/cypress/integration/subproject_edit_spec.js]: true, I added the check

File: e2e-test/cypress/integration/project_edit_spec.js (hunk 3 above)
Comment (Stezido) [reply e2e-test/cypress/integration/project_edit_spec.js]: Done. Since most tests are done with user "mstein" should we remove all view permissions on projects created in e2e-test phase?

Comment (mathiashoeld) [reply e2e-test/cypress/integration/project_edit_spec.js]: no, since the tests are usually run in a separate container in the CI/CD pipeline this is not a very big issue. users testing locally can perform this action manually if they like
